File: app/src/main/java/com/example/notphilphil/bob/models/Location.java
package com.example.notphilphil.bob.models;

import java.util.Arrays;

public class Location {
    private String Zip;
    private String Type;
    private String Phone;
    private String State;
    private String StreetAddress;
    private String Website;
    private double Latitude;
    private double Longitude;
    private String City;
    private String Name;

    public static final String[] tokens = {"Latitude", "Longitude", "Street Address", "City", "State", "Zip", "Type", "Phone", "Website", "Name"};
    private String Key;

    private Location(String zip, String type, String phone, String state, String streetAddress, String website, String city, String name, String key, String lat, String lon) {
        Zip = zip;
        Type = type;
        Phone = phone;
        State = state;
        StreetAddress = streetAddress;
        Website = website;
        Latitude = Double.parseDouble(lat);
        Longitude = Double.parseDouble(lon);
        City = city;
        Name = name;
        Key = key;
    }

    public Location() {

    }

    public String getZip() {
        return Zip;
    }

    private void setZip(String zip) {
        Zip = zip;
    }

    public String getType() {
        return Type;
    }

    private void setType(String type) {
        Type = type;
    }

    public String getPhone() {
        return Phone;
    }

    private void setPhone(String phone) {
        Phone = phone;
    }

    public String getState() {
        return State;
    }

    private void setState(String state) {
        State = state;
    }

    public String getStreetAddress() {
        return StreetAddress;
    }

    private void setStreetAddress(String streetAddress) {
        StreetAddress = streetAddress;
    }

    public String getWebsite() {
        return Website;
    }

    private void setWebsite(String website) {
        Website = website;
    }

    public double getLatitude() {
        return Latitude;
    }

    private void setLatitude(double latitude) {
        Latitude = latitude;
    }

    public double getLongitude() {
        return Longitude;
    }

    private void setLongitude(double longitude) {
        Longitude = longitude;
    }

    public String getCity() {
        return City;
    }

    private void setCity(String city) {
        City = city;
    }

    public String getName() {
        return Name;
    }

    private void setName(String name) {
        Name = name;
    }

    public String getKey() {
        return Key;
    }

    public void setKey(String key) {
        Key = key;
    }

    public void addValue(String key, String data) {
        switch (Arrays.asList(tokens).indexOf(key)) {
            case 0: this.setLatitude(Double.parseDouble(data)); break;
            case 1: this.setLongitude(Double.parseDouble(data)); break;
            case 2: this.setStreetAddress(data); break;
            case 3: this.setCity(data); break;
            case 4: this.setState(data); break;
            case 5: this.setZip(data); break;
            case 6: this.setType(data); break;
            case 7: this.setPhone(data); break;
            case 8: this.setWebsite(data); break;
            case 9: this.setName(data); break;
        }
    }

    @Override
    public String toString() {
        return
            "Latitude: " + this.Latitude +
            "\nLongitude: " + this.Longitude+
            "\nStreet Address: " + this.StreetAddress +
            "\nCity: " + this.City +
            "\nState: " + this.State +
            "\nZip: " + this.Zip +
            "\nType: " + this.Type +
            "\nPhone: " + this.Phone +
            "\nWebsite: " + this.Website;
    }
}

File: app/src/main/java/com/example/notphilphil/bob/models/Location.kt
package com.example.notphilphil.bob.models

import java.util.Arrays

class Location {
    var zip: String? = null
        private set
    var type: String? = null
        private set
    var phone: String? = null
        private set
    var state: String? = null
        private set
    var streetAddress: String? = null
        private set
    var website: String? = null
        private set
    var latitude: Double = 0.toDouble()
        private set
    var longitude: Double = 0.toDouble()
        private set
    var city: String? = null
        private set
    var name: String? = null
        private set
    var key: String? = null

    private constructor(zip: String, type: String, phone: String, state: String, streetAddress: String, website: String, city: String, name: String, key: String, lat: String, lon: String) {
        this.zip = zip
        this.type = type
        this.phone = phone
        this.state = state
        this.streetAddress = streetAddress
        this.website = website
        latitude = java.lang.Double.parseDouble(lat)
        longitude = java.lang.Double.parseDouble(lon)
        this.city = city
        this.name = name
        this.key = key
    }

    constructor() {

    }

    fun addValue(key: String, data: String) {
        when (Arrays.asList(*tokens).indexOf(key)) {
            0 -> this.latitude = java.lang.Double.parseDouble(data)
            1 -> this.longitude = java.lang.Double.parseDouble(data)
            2 -> this.streetAddress = data
            3 -> this.city = data
            4 -> this.state = data
            5 -> this.zip = data
            6 -> this.type = data
            7 -> this.phone = data
            8 -> this.website = data
            9 -> this.name = data
        }
    }

    override fun toString(): String {
        return "Latitude: " + this.latitude +
                "\nLongitude: " + this.longitude +
                "\nStreet Address: " + this.streetAddress +
                "\nCity: " + this.city +
                "\nState: " + this.state +
                "\nZip: " + this.zip +
                "\nType: " + this.type +
                "\nPhone: " + this.phone +
                "\nWebsite: " + this.website
    }

    companion object {

        val tokens = arrayOf("Latitude", "Longitude", "Street Address", "City", "State", "Zip", "Type", "Phone", "Website", "Name")
    }
}
